Add view for 'no projects yet' -page

PatricKangasmaki opened the pull request:
There's no users without roadmaps currently so to test this, you need to remove someone from their roadmaps and then log in. Create a new project button doesn't do anything yet, it needs to be updated later when the modal is ready.

IidaKainu commented on frontend/src/pages/CreateProjectPage.tsx:
```diff
@@ -0,0 +1,55 @@
+            <Trans i18nKey="First project info 1/2" />
+          </div>
+          <div className={classes(css.body)}>
+            <Trans i18nKey="First project info 1/2" />
```
Looks like a typo: the first info is repeated twice
```suggestion
            <Trans i18nKey="First project info 2/2" />
```

IidaKainu reviewed (changes requested):
Looks good!
There's some similar css in create project, login and register pages (white container/formDiv, footer). They could be combined 👍

PatricKangasmaki commented:
Fixed the typo and moved css-classes that were used in multiple places to the shared.scss folder 👍 I left the footer in the createProject-page module untouched since it has its own translations and positions

juhoeerola commented on frontend/src/components/LoginNavBar.tsx:
```diff
@@ -10,26 +10,37 @@ import css from './LoginNavBar.module.scss';
       </Link>
-    ) : (
+    )}
+    {type === 'landing' && (
+      <Link className={classes(css.loginButton)} to={paths.loginPage}>
+        <Trans i18nKey="Login" />
+      </Link>
+    )}
+    {type === 'register' && (
```
These could maybe be combined
```suggestion
    {(type === 'register' || type === 'landing') && (
```

juhoeerola commented on frontend/src/components/NavBar.tsx:
```diff
@@ -31,26 +31,29 @@ export const NavBar = () => {
-                <PowerSettingsNewIcon className={classes(css.icon)} />
-              </Link>
-            </>
+      {userInfo &&
```
I think this whole component could be refactored a bit

Using couple conditional early returns would make this much clearer in my opinion, e.g.

```ts
  if (!userInfo) {
    return pathname.startsWith(paths.loginPage) ? (
      <LoginNavBar type="login" />
    ) : (
      <LoginNavBar type="register" />
    );
  }

  if (pathname.startsWith(paths.getStarted))
    return <LoginNavBar type="getStarted" />;

  return (
    <div className={classes(css.navBarDiv)}>
      <CornerPiece />
      <div className={classes(css.navBarRightSide)}>
        <div className={classes(css.navBarText)}>
          <Trans i18nKey="Project" />
        </div>
        <RoadmapSelectorWidget />
        <div className={classes(css.navBarDivider)} />
        <Link className={classes(css.navBarLink)} to={paths.userInfo}>
          <PermIdentityIcon className={classes(css.icon)} />
        </Link>
        <Link className={classes(css.navBarLink)} to={paths.logoutPage}>
          <PowerSettingsNewIcon className={classes(css.icon)} />
        </Link>
      </div>
    </div>
  );
```

PatricKangasmaki commented:
Added all your suggestions!

juhoeerola reviewed (approved):
Look good 👍 

Which ever is merged later, this or #116, can add the functionality to the "Create new project" button

IidaKainu reviewed (approved):
👍
